File: src/main/com/dune/DuneBundle.java
package com.dune;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

/**
 * Localization
 */
public class DuneBundle extends DynamicBundle {

    @NonNls
    public static final String BUNDLE = "messages.DuneBundle";
    private static final DuneBundle INSTANCE = new DuneBundle();

    private DuneBundle() {
        super(BUNDLE);
    }

    @NotNull
    public static @Nls String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key,
                                      Object ... params) {
        return INSTANCE.getMessage(key, params);
    }
}

File: src/main/kotlin/com/dune/DuneBundle.kt
package com.dune

import com.intellij.DynamicBundle
import org.jetbrains.annotations.PropertyKey

private const val BUNDLE = "messages.DuneBundle"

// ex: DuneBundle.message("filetype.dune.description")
object DuneBundle : DynamicBundle(BUNDLE) {
    fun message(@PropertyKey(resourceBundle = BUNDLE) key: String,
                vararg params: Any): String {
        return getMessage(key, *params)
    }
}